qiskit_nature/second_q/mappers/bosonic_logarithmic_mapper.py

```diff
@@ -0,0 +1,266 @@
+                        # qubit register.
+                        i: int = len(init_state) - j - 1
+                        # Case |0><0|: this should be converted to 0.5*(I + Z)
+                        if f"{final_state[j]}{init_state[j]}" == "00":
+                            single_mapped_term = single_mapped_term.compose(
+                                self._get_single_qubit_pauli_matrix(
+                                    mode_index_in_register, qubit_register_length, i, "I+"
+                                )
+                            )
+                        # Case |1><1|: this should be converted to 0.5*(I - Z)
+                        elif f"{final_state[j]}{init_state[j]}" == "11":
+                            single_mapped_term = single_mapped_term.compose(
+                                self._get_single_qubit_pauli_matrix(
+                                    mode_index_in_register, qubit_register_length, i, "I-"
+                                )
+                            )
+                        # Case |0><1|: this should be converted to 0.5*(X + iY)
+                        elif f"{final_state[j]}{init_state[j]}" == "01":
+                            single_mapped_term = single_mapped_term.compose(
+                                self._get_single_qubit_pauli_matrix(
+                                    mode_index_in_register, qubit_register_length, i, "S+"
+                                )
+                            )
+                        # Case |1><0|: this should be converted to 0.5*(X - iY)
+                        elif f"{final_state[j]}{init_state[j]}" == "10":
+                            single_mapped_term = single_mapped_term.compose(
+                                self._get_single_qubit_pauli_matrix(
+                                    mode_index_in_register, qubit_register_length, i, "S-"
+                                )
+                            )
+                        else:
+                            raise ValueError(f"Invalid state {final_state[i]}{init_state[i]}.")
+                    pauli_expansion.append(prefactor * single_mapped_term)
+                # Add the Pauli expansion for a single n_k to map of the bosonic operator
+                bos_op_to_pauli_op = reduce(operator.add, pauli_expansion).compose(
+                    bos_op_to_pauli_op
+                )
+            # Add the map of the single boson op (e.g. +_0) to the map of the full bosonic operator
+            pauli_op.append(coeff * reduce(operator.add, bos_op_to_pauli_op.simplify()))
+        # return the lookup table for the transformed XYZI operators
+        return reduce(operator.add, pauli_op)
+
+    @lru_cache(maxsize=32)
+    def _get_binary_state(self, n: int) -> str:
+        """This method converts an integer to its binary representation."""
+        return bin(n).split("b")[1].rjust(self.number_of_qubits_per_mode, "0")
```

Comment:
I think this f-string is much simpler:
```suggestion
        return f"{n:0{self.number_of_qubits_per_mode}b}"
```
At that point you can even consider removing this extra function and caching really becomes unnecessary on it.

Reply:
Thanks for this tip, it works and it;s definitely much nicer :)